pkg/plugins/bootstrap/k8s/plugin.go

```diff
@@ -48,7 +45,9 @@ func (p *plugin) Bootstrap(b *core_runtime.Builder, _ core_plugins.PluginConfig)
 		return err
 	}
 	b.WithComponentManager(&kubeComponentManager{mgr})
-	b.WithExtensions(k8s_runtime.NewManagerContext(b.Extensions(), mgr))
+	b.WithExtensions(k8s_extensions.NewManagerContext(b.Extensions(), mgr))
+	b.WithExtensions(k8s_extensions.NewResourceConverterContext(b.Extensions(), k8s.NewCachingConverter(5*time.Minute)))
```

Comment:
I would feel slightly better about having this value configurable (and disable on 0s), just in case so if something goes wrong, we can turn this off.

Reply:
Yes, that makes sense, fixed